Update docs for pluggin in CA certs (#8099)
istio/istio.io: content/en/docs/tasks/security/cert-management/plugin-ca-cert/index.md

---
title: Plug in CA Certificates
description: Shows how system administrators can configure Istio's CA with a root certificate, signing certificate and key.
weight: 80
keywords: [security,certificates]
aliases:
    - /docs/tasks/security/plugin-ca-cert/

> test: true

Reviewer: Test needs to be updated.

Author: @frankbu there is no test currently ... do you mean snips.sh? It's updated now.

Author: oh nm ... looks like it was just checked in.  I see it now ... will update.

---

> This task shows how administrators can configure the Istio certificate authority with a root certificate,

Reviewer: ```suggestion
This task shows how administrators can configure the Istio certificate authority (CA) with a root certificate,
```

signing certificate and key.

By default, Istio's CA generates a self-signed root certificate and key, and uses them to sign the workload certificates.
Istio's CA can also sign workload certificates using an administrator-specified certificate and key, and with an
administrator-specified root certificate.

> The root CA is used by all workloads within the mesh as the root of trust. Each Istio CA uses an intermediate CA
> signing key and certificate, signed by the root CA. When multiple Istio CAs exist within the mesh, this establishes a
> hierarchy of trust among the CAs.

Reviewer: ```suggestion
A root CA is used by all workloads within a mesh as the root of trust. Each Istio CA uses an intermediate CA
signing key and certificate, signed by the root CA. When multiple Istio CAs exist within a mesh, this establishes a
hierarchy of trust among the CAs.
```

> {{< image width="80%"

Reviewer: Diagram is too small and surrounded by whitespace. See preview.

    link="hierarchical-pki.png"

>     caption="Hierarchical PKI"

Reviewer: I wouldn't use acronym `PKI` in the caption, unless it appears in (and is expanded) in the body text as well.

    >}}

> This task demonstrates how to generate and plug in certificates and key into Istio's CA. These steps can be repeated

Reviewer: ```suggestion
This task demonstrates how to generate and plug in the certificates and key for Istio's CA. These steps can be repeated
```

to provision certificates and keys for any number of Istio CAs.

## Plug in certificates and key into the cluster

{{< tip >}}
For production cluster setup, it is a good practice to do the following on an offline machine with good

> security protection. The root private key should be exposed to minimal people and processes.

Reviewer: ```suggestion
security protection. The root private key should be exposed to as few people and processes as possible.
```

1.  Verify the certificate chain from the root certificate to the workload certificate:

    {{< text bash >}}
    $ openssl verify -CAfile <(cat samples/certs/cluster1/ca-cert.pem samples/certs/cluster1/root-cert.pem) ./proxy-cert-1.pem
    ./proxy-cert-1.pem: OK
    {{< /text >}}

## Cleanup

> *   To remove the certificates, keys, and intermediate files from your local disk:

Reviewer: ```suggestion
*   Remove the certificates, keys, and intermediate files from your local disk:
```

    {{< text bash >}}
    $ rm -rf cluster1
    {{< /text >}}

> *   To remove the secret `cacerts`, and the `foo` and `istio-system` namespaces:

Reviewer: ```suggestion
*   Remove the secret `cacerts`, and the `foo` and `istio-system` namespaces:
```
